src/caffeine/prif_queries_s.f90

```diff
@@ -13,6 +18,16 @@
   end procedure
 
   module procedure prif_base_pointer
+    integer(c_int) :: num_img
+
+    call prif_num_images(image_count=num_img)
+    call assert(image_num .gt. 0 .or. image_num .lt. num_img, "prif_base_pointer: image_num not within valid range")
+    if (image_num .eq. 0) then
+      ptr = 0
```

Comment:
is it true that (in C) `(intptr_t)NULL == 0`? I'm pretty sure it would definitely be right as `ptr = as_int(c_null_ptr)`

Reply:
Yes, the C standard guarantees null pointers have the value zero.
6.3.2.3:
> An integer constant expression with the value 0, or such an expression cast to type void *, is called a null pointer constant.66) If a null pointer constant is converted to a pointer type, the resulting pointer, called a null pointer, is guaranteed to compare unequal to a pointer to any object or function.